site/docs/concepts/schemas.md

```diff
@@ -211,6 +214,42 @@ but you can also use absolute URLs to a third-party schema like
 [schemastore.org](https://www.schemastore.org).
 :::
 
+## Write and read schemas
+
+In some cases, you may want to impose different constraints to data that is entering (_written to_) the collection
+and data that is exiting (_read from_) the collection.
+
+For example, you may need to start capturing data _now_ from a source system; say, a pub-sub system with short-lived
+historical data support or an HTTP endpoint, but don't have time to construct the ideal schema you'll need later.
+You can capture the data with a permissive write schema, and impose a stricter read schema on the data
+as you need to perform a derivation or materialization.
+You can safely experiment with the read schema at your convenience, knowing the data has already been captured.
+
+To achieve this, edit the collection, re-naming the standard `schema` to `writeSchema` and adding a `readSchema`.
+You can also supply a `writeSchema` in place of `schema` at capture time; it will be used for both writes and reads until a `readSchema` is added.
+
+Here's a simple example in which you don't know how purchase prices are formatted when capturing them,
+but find out later that `number` is the appropriate data type:
+
+```yaml
+collections:
+  purchases:
+    writeSchema:
+      type: object
+      title: Store price as strings
+      description: Not sure if prices contain characters such as $, so capturing them as strings.
+      properties:
+        id: { type: integer}
+        price: {type: string}
+    readSchema:
+      type: object
+      title: Prices as numbers
+      properties:
+        id: { type: integer}
+        price: {type: number}
+    key: [/id]
+```
```

Comment:
Other bits of further detail to perhaps note:

* The collection key must be defined by both the write and read schema.

* Projections of the collection generally must only exist / be defined by the read schema, but _logical partitions_ must additionally be defined by the write schema.